test/integration/http_server_action_test.lua

```diff
@@ -0,0 +1,107 @@
+local fio = require('fio')
+local fiber = require('fiber')
+local socket = require('socket')
+local t = require('luatest')
+local http_client = require('http.client')
+local http_server = require('http.server')
+
+local helpers = require('test.helpers')
+
+local g = t.group()
+
+g.before_test('test_stop_basic', function()
+    g.base_host = '127.0.0.1'
+    g.base_port = 8080
+    g.httpd = http_server.new(g.base_host, g.base_port, {
+        app_dir = helpers.path,
+    })
+    g.httpd:start()
+end)
+
+g.after_test('test_stop_basic', function()
+    helpers.retrying({}, function()
+        local conn = socket.tcp_connect(g.base_host, g.base_port)
+        return conn ~= nil
+    end)
+    fio.rmdir(helpers.path)
+end)
+
+g.before_test('test_stop_with_open_connection', function()
+    g.base_host = '127.0.0.1'
+    g.base_port = 8080
+    g.base_uri = ('http://%s:%s'):format(g.base_host, g.base_port)
+    g.httpd = http_server.new(g.base_host, g.base_port, {
+        app_dir = helpers.path,
+    })
+    g.httpd:start()
+end)
+
+g.after_test('test_stop_with_open_connection', function()
+    helpers.retrying({}, function()
```

Comment:
`after_test` is for cleanup. If we want to assert some logic, I think we should do it in the main test body.